### PR TITLE
add user story 12

### PR DESCRIPTION
**Pull request by jhgould**
Adds user story 12

### CODE REVIEW

File: spec/features/user/index_spec.rb [hunk 1]
```diff
@@ -1,60 +1,60 @@
...
-#   fill_in :zip, with: 88832
-#   fill_in :email, with: "bob@gz.com"
-#   fill_in :password_confirmation, with: "1234"
-#   click_on "Create User"
-#   expect(current_path).to eql("/register")
-#
-#   expect(page).to have_content("Password can't be blank")
-# end
+  it "Can register a user" do
+    visit "/merchants"
```
Comment (EricLarson2020) [inline spec/features/user/index_spec.rb:5]: Nice Work, looks good here

File: spec/features/user/index_spec.rb [hunk 2]
```diff
@@ -73,6 +73,40 @@
...
+    fill_in :address, with: "333 Blvd"
+    fill_in :city, with: "Denver"
+    fill_in :state, with: "Colorado"
+    fill_in :zip, with: 88832
+    fill_in :email, with: "jack@hotmail.com"
+    fill_in :password, with: "1234"
+    fill_in :password_confirmation, with: "1234"
+    click_button "Create User"
+    expect(current_path).to eql("/register")
+    expect(page).to have_content("Email has already been taken")
```
Comment (whitneykidd) [inline spec/features/user/index_spec.rb:106]: This error message is super sleek

File: spec/features/user/index_spec.rb [hunk 3]
```diff
@@ -73,6 +73,40 @@
     expect(current_path).to eq("/register")
     expect(page).to have_content("Password confirmation doesn't match Password")
   end
+
+  it "user has to use unique email" do
```
Comment (EricLarson2020) [inline spec/features/user/index_spec.rb:77]: Nice Work

File: spec/features/user/index_spec.rb (hunk 2 above)
Comment (jhgould) [reply spec/features/user/index_spec.rb:106]: Gotta love rails magic

**Pull request merged**